components/ui-api-layer/docs/resource-quotas-status.md

```diff
@@ -0,0 +1,116 @@
+# ResourceQuotasStatus 
+
+[ResourceQuota](https://kubernetes.io/docs/concepts/policy/resource-quotas/) provides constraints that limit aggregate resource consumption per Namespace.
+
+The ResourceQuota values can be as follows:
+```bash
+Name:            kyma-default
+Resource         Used       Hard
+--------         ---        ---
+limits.memory    7348440Ki  10Gi
+requests.memory  3727576Ki  7Gi
+```
+The `used` values of the ResourceQuota can exceed the `hard` values and block any resource creation in the Environment.
+
+The ResourceQuotasStatus contains the flag and list of exceeded ResourceQuotas limits, together with a set of resources which exceed that limits. The value of the flag informs you if any ResourceQuota is exceeded in any possible way.
+
+ResourceQuotasStatus should detect whenever any ReplicaSet or StatefulSet in your Environment cannot progress because it's blocked by the ResourceQuota.
+To check that you must calculate the required amount of resources to create another replica. If there is not enough resources, you should return the exceeded status.
+
+[Here](https://github.com/dtaylor113/origin-web-console/blob/master/app/scripts/services/quota.js#L86) you can find the example implementation of the checking logic in UI.
+
+## The Console calls
+
+The Console calls for the ResourceQuotasStatus automatically. The calls are triggered after: 
+- switching the environment
+- uploading a resource
+- creating a lambda 
+- opening environment's details
+
+## Implementation
+
+This section contains the steps of the ResourceQuotaStatus implementation.
+
+### Calculate the available resources in the Environment
+
+To calculate amount of available resources in the given Environment you must list the ResourceQuotas and loop through them.
+You can get the available amount of resources by calculating the difference from `.spec.hard` and `.status.used` in each ResourceQuota.
+For each resource limit specified in the ResourceQuotas, you must calculate the available amount of resources using the ResourceQuota with the lowest `.spec.hard` value for that resource type.
+
+### Calculate the necessary resources to create the next replica
+
+To calculate how much resources you need to create the next replica, you must sum up the resource usage of all containers in the replica Pod.
+If some resource limit is not specified directly in the replica's template, you must check if any LimitRange is not specifying it.
+In this situation you must add the proper LimitRange limit to the list of necessary resources.
```

Comment:
`In this situation, add the proper LimitRange limit to the list of necessary resources.`